src/runtime_filesystem.cpp

```
@@ -162,42 +162,64 @@ int64_t __ry_filesystem_copy(const char *src, const char *dst) {
-    FILE *in = fopen(src, "rb");
-    if (!in) {
-        setLastError("copy: cannot open source '%s': %s", src, strerror(errno));
+    struct stat src_st;
+    if (fstat(src_fd, &src_st) != 0) {
+        setLastError("copy: cannot stat source '%s': %s", src, strerror(errno));
+        close(src_fd);
         return 1;
     }
-    FILE *out = fopen(dst, "wb");
-    if (!out) {
+    int dst_fd = open(dst, O_WRONLY | O_CREAT | O_TRUNC, src_st.st_mode & 07777);
```

Clang (14.0.6)

[warning] variable 'dst_fd' is not initialized

(cppcoreguidelines-init-variables)